SQL_Solution.txt

```diff
@@ -0,0 +1,1679 @@
+| CNAME    | ODATE      |
++----------+------------+
+| Cisneros | 1990-03-10 |
+| Pereira  | 1990-03-10 |
+| Hoffman  | 1990-03-10 |
+| Liu      | 1990-03-10 |
+| Cisneros | 1990-03-10 |
++----------+------------+
+
+15) Give the sums of the amounts from the Orders table, grouped by date, eliminating all those dates where the SUM was not at least 2000 above the maximum Amount. 
+mysql> SELECT SUM(AMT),ODATE
+    ->     FROM ORDERS
+    ->     WHERE AMT > 2000 GROUP BY ODATE;
++----------+------------+
+| SUM(AMT) | ODATE      |
++----------+------------+
+|  5160.45 | 1990-03-10 |
+|  4723.00 | 1990-05-10 |
+|  9891.88 | 1990-06-10 |
++----------+------------+
+
+16) Select all orders that had amounts that were greater than at least one of the orders from October 6. 
+mysql> SELECT ODATE,ONUM,AMT
+    ->     FROM ORDERS WHERE
+    ->     AMT >= ANY (
+    ->         SELECT AMT FROM ORDERS WHERE ODATE = "1990-06-10"
+    -> );
++------------+------+---------+
+| ODATE      | ONUM | AMT     |
++------------+------+---------+
+| 1990-03-10 | 3002 | 1900.10 |
+| 1990-03-10 | 3005 | 5160.45 |
+| 1990-05-10 | 3008 | 4723.00 |
+| 1990-04-10 | 3009 | 1713.23 |
+| 1990-06-10 | 3010 | 1309.95 |
+| 1990-06-10 | 3011 | 9891.88 |
++------------+------+---------+
+
+
+17) Write a query that uses the EXISTS operator to extract all salespeople who have customers with a rating of 300.
+mysql> SELECT SNUM,SNAME
+    ->     FROM SALESPEOPLE S WHERE EXISTS  (
+    ->         SELECT *
+    ->         FROM CUSTOMERS C
+    ->         WHERE C.SNUM = S.SNUM AND RATING = 300
+    ->     );
++------+--------+
+| SNUM | SNAME  |
++------+--------+
+| 1002 | Serres |
+| 1007 | Rifkin |
++------+--------+
+
+ 18) Find all customers whose cnum is 1000 above the snum of Serres. 
+mysql> SELECT C.CNAME,C.CNUM,S.SNAME,S.SNUM FROM CUSTOMERS C
+    ->     JOIN SALESPEOPLE S ON
+    ->     C.SNUM = S.SNUM WHERE
+    ->     CNUM > 1002 + 1000;
++----------+------+--------+------+
+| CNAME    | CNUM | SNAME  | SNUM |
++----------+------+--------+------+
+| Liu      | 2003 | Serres | 1002 |
+| Grass    | 2004 | Serres | 1002 |
+| Clemens  | 2006 | Peel   | 1001 |
+| Pereira  | 2007 | Motika | 1004 |
+| Cisneros | 2008 | Rifkin | 1007 |
++----------+------+--------+------+
+
+19) Give the salespeople’s commissions as percentages instead of decimal numbers. 
+ SELECT SNUM,SNAME,(COMM/(SELECT SUM(COMM) FROM SALESPEOPLE))*100 + '%'  AS COMMISION FROM SALESPEOPLE;
++------+---------+------------+
+| SNUM | SNAME   | COMMISION  |
++------+---------+------------+
+| 1001 | Peel    | 13.9534883 |
+| 1002 | Serres  |  15.116279 |
+| 1003 | AxelRod | 11.6279069 |
+| 1004 | Motika  | 12.7906976 |
+| 1007 | Rifkin  | 17.4418604 |
+| 1008 | Fran    | 29.0697674 |
++------+---------+------------+
+
+20) Find the largest order taken by each salesperson on each date, eliminating those Maximum orders, which are less than 3000. 21) List all the largest orders for October 3, for each salesperson. 
```

Comment:
what is this? when you are completing this ,any problem?

Reply:
Resolved